dynamic_programming/trapped_rainwater.cpp

```diff
@@ -0,0 +1,101 @@
+/**
+ * @file
+ * @brief Implementation of the [Trapped Rainwater
+ * Problem](https://www.geeksforgeeks.org/trapping-rain-water/)
+ * @details
+ * This implementation calculates the amount of rainwater that can be trapped
+ * between walls represented by an array of heights.
+ */
+
+#include <algorithm>  /// For std::min and std::max
+#include <cassert>    /// For assert
+#include <vector>     /// For vector container
+
+/*
+ * @namespace
+ * @brief Dynamic Programming Algorithms
+ */
+namespace dynamic_programming {
+/**
+ * @brief Function to calculate the trapped rainwater
+ * @param heights Array representing the heights of walls
+ * @return The amount of trapped rainwater
+ */
+int trappedRainwater(const std::vector<int>& heights) {
```

Comment:
should this not be some form of unsigned int?
since heights cant be negative the return also could be unsigned

Reply:
Yes, It should be unsigned int